src/Pond.jl

```diff
@@ -0,0 +1,119 @@
+module Pond
+using ClimaLSM
+using ClimaLSM.Domains: AbstractDomain
+using ClimaLSM.ComponentExchanges: AbstractComponentExchange
+using ClimaCore
+using UnPack
+import ClimaLSM: AbstractModel, make_rhs, prognostic_vars
+
+export PondModel, PondPrescribedExchange
+
+"""
+    PondModel{FT, D, B} <: AbstractModel{FT}
+
+A stand-in model for models like the snow or river model. In 
+standalone mode, soil conditions underneath the pond and precipitation
+control the rate of change of the pond height variable `η` via an ODE.
+For a true river model, this computed term would be a source term.
+
+In integrated LSM mode, the infiltration into the soil will be computed
+via a different method, and also be applied as a flux boundary condition
+for the soil model. 
+"""
+struct PondModel{FT, D, B} <: AbstractModel{FT}
+    domain::D
+    boundary_exchanges::B
+    model_name::Symbol
+end
+
+function PondModel{FT}(;
+    domain::AbstractDomain{FT},
+    boundary_exchanges::AbstractComponentExchange{FT},
+) where {FT}
+    return PondModel{FT, typeof(domain), typeof(boundary_exchanges)}(
+        domain,
+        boundary_exchanges,
+        :pond,
+    )
+end
+"""
+    PondPrescribedExchange{FT} <: AbstractComponentExchange{FT}
+
+The required input for driving the simple pond model: precipitation, as a
+function of time, soil effective saturation at a depth `Δz` below the surface,
+as a function of time, and soil parameters, which affect infiltration.
+"""
+struct PondPrescribedExchange{FT} <: AbstractComponentExchange{FT}
+    "Time dependent precipitation < 0 , given in m/s"
+    P::Function
+    "Time dependent soil effective saturation at depth z_sfc-Δz, ∈ (0,1]"
+    S::Function
+    "Saturated conductivity Ksat (m/s), van Genuchten parameters α (1/m), n, and Δz (m)"
+    soil_params::Tuple{FT, FT, FT, FT}
+end
+
+prognostic_vars(model::PondModel) = (:η,)
+
+function make_rhs(model::PondModel)
+    function rhs!(dY, Y, p, t)
+        precip = compute_precipitation(model.boundary_exchanges, Y, p, t)
+        infiltration = compute_infiltration(model.boundary_exchanges, Y, p, t)
+        dY.pond.η .= precip - infiltration
+    end
+    return rhs!
+end
+
+
+infiltration_at_point(η::FT, i_c::FT, P::FT) where {FT} =
+    η > eps(FT) ? i_c : min(i_c, P)
+
+
+
+
+function compute_infiltration(
+    boundary_exchanges::PondPrescribedExchange{FT},
+    Y::ClimaCore.Fields.FieldVector,
+    p::ClimaCore.Fields.FieldVector,
+    t::FT,
+)::FT where {FT}
+
+    S = boundary_exchanges.S(t)
+    Ksat, α, n, Δz = boundary_exchanges.soil_params
+    m = FT(1) - FT(1) / n
+    i_c =
+        hydraulic_conductivity(Ksat, m, S) .*
+        (max.(FT(0.0), Y.pond.η) .- matric_potential(α, n, m, S) .+ Δz) ./ Δz
+    println(typeof(i_c))
+    println(typeof(Y.pond.η))
+    println(typeof(boundary_exchanges.P(t)))
+    infiltration =
```

Comment:
@jb-mackay This was an example where I had to use `Y.pond.eta[1]` a lot originally. 
```
 i_c =
        hydraulic_conductivity(Ksat, m, S) *
        (max(FT(0.0), Y.pond.η[1]) - matric_potential(α, n, m, S) + Δz) / Δz
    if Y.pond.η[1] > eps(FT) # when there is a pond, infiltration is pond/soil limited
        infiltration = i_c
    else # when there is no pond, infiltration is precip/soil limited
        infiltration = min(i_c, boundary_exchanges.P(t))
    end
```

But I tried to rewrite it so we do not...it seems like it should work, but this line is leading to errors - only when run within the function.  calling `infiltration_at_point` by itself with objects of the same type `infiltration_at_point.(a::Vector{FT}, b::Vector{FT}, c::FT)` does not produce an error.

Reply:
fixed this! all set